mobile/lib/modules/backup/providers/backup.provider.dart

```diff
@@ -267,48 +258,29 @@ class BackupNotifier extends StateNotifier<BackUpState> {
       }
       AssetPathEntity albumHasAllAssets = list.first;
 
-      backupAlbumInfoBox.put(
-        backupInfoKey,
-        HiveBackupAlbums(
-          selectedAlbumIds: [albumHasAllAssets.id],
-          excludedAlbumsIds: [],
-          lastSelectedBackupTime: [
-            DateTime.fromMillisecondsSinceEpoch(0, isUtc: true)
-          ],
-          lastExcludedBackupTime: [],
-        ),
+      final ba = BackupAlbum(
+        albumHasAllAssets.id,
+        DateTime.fromMillisecondsSinceEpoch(0),
+        BackupSelection.select,
       );
-
-      backupAlbumInfo = backupAlbumInfoBox.get(backupInfoKey);
+      await _db.writeTxn(() => _db.backupAlbums.put(ba));
     }
 
     // Generate AssetPathEntity from id to add to local state
     try {
       Set<AvailableAlbum> selectedAlbums = {};
-      for (var i = 0; i < backupAlbumInfo!.selectedAlbumIds.length; i++) {
-        var albumAsset =
-            await AssetPathEntity.fromId(backupAlbumInfo.selectedAlbumIds[i]);
+      for (BackupAlbum ba in selectedBackupAlbums) {
+        var albumAsset = await AssetPathEntity.fromId(ba.id);
```

Comment:
Probably make `final` instead of `var`.

Reply:
Oh, and `ba` can be `final` as well in the `for` loop.

Reply:
didn't know you can apply final in a loop variable in dart. thanks for the pointer!